crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -3575,9 +3489,10 @@ impl AuthorityPerEpochStore {
 
     pub(crate) fn write_pending_checkpoint(
         &self,
-        batch: &mut DBBatch,
+        output: &mut ConsensusCommitOutput,
         checkpoint: &PendingCheckpointV2,
     ) -> SuiResult {
+        // TODO: is this check necessary?
```

Comment:
it seems like not? I can't see how multiple pending checkpoints could be written at the same height. at the very least it could be simplified to `assert!(self.get_pending_checkpoint(&checkpoint.height())?.is_none())`

Reply:
i did try to find test failures when turning this into a panic, and couldn't: https://github.com/MystenLabs/sui/pull/18432

replacing with an assert